### PR TITLE
Medical - Prevent running `FUNC(deserializeState)` in scheduled environment

### LINKED ISSUE #9636: undefined variable error in fnc_deserializeState.sqf at line 54. Found a fix. Might need to be added to repository
variable name "_default" = undefined variable error in file _**fnc_deserializeState.sqf**_  in medical addon.

Seems to be in the loop at line 54 (below comment _// Set medical variables_) and is thrown when the _default value is null in the loop. This loop is processing values in the hashmap. 

Example hashmap (formatted to json) to test:
```
"{""ace_medical_openwounds"": {""rightleg"": [[61, 2, 0.0818448, 0.796858], [11, 2, 0.0426049, 0.753706]]}, ""ace_medical_bloodpressure"": [93, 139], ""ace_medical_ivbags"": null, ""ace_medical_inpain"": true, ""ace_medical_medications"": [], ""ace_medical_tourniquets"": [0, 0, 0, 0, 0, 0], ""ace_medical_heartrate"": 101.015, ""ace_medical_pain"": 0.419046, ""ace_medical_bloodvolume"": 5.7542, ""ace_medical_stitchedwounds"": {}, ""ace_medical_fractures"": [0, 0, 0, 0, 0, 1], ""ace_medical_triagelevel"": 0, ""ace_medical_bodypartdamage"": [0, 0, 0, 0, 0, 3.10113], ""ace_medical_hemorrhage"": 1, ""ace_medical_occludedmedications"": null, ""ace_medical_triagecard"": [], ""ace_medical_bandagedwounds"": {}, ""ace_medical_peripheralresistance"": 100, ""ace_medical_painsuppress"": 0, ""ace_medical_statemachinestate"": ""Injured""}"
```

it throws the error on any key with a null value, so "ace_medical_ivbags", "ace_medical_occludedmedications" throw the error.

To get the hashmap I used
```sqf
private _jsonStr = _unit call ace_medical_fnc_serializeState; 							
private _jsonhash = [_jsonStr, 2] call CBA_fnc_parseJSON;  // 2nd arg will get native hashMaps
```

and sending back to ACE with (which throws the error)

```sqf
private _newJsonStr  = [_jsonhash] call CBA_fnc_encodeJSON;
[_unit, _newJsonStr] call ace_medical_fnc_deserializeState;	
```
						
The error doesnt seem to affect functionality, it just throws annoying error in debugger.

Cheers

Ben
						

**Mods **
- **Arma 3:** `2.14` [e.g. 1.00 stable, rc,  dev]
- **CBA:** `3.16.1` [e.g. 3.0.0 stable, commit hash]
- **ACE3:** `3.16.1` [eg. 3.0.0 stable, commit hash]

occurs:
-  Editor (Multiplayer) 

link to error line and file:
https://github.com/acemod/ACE3/blame/6bc5193a79e0388e908441e34fb82b45e8359b7c/addons/medical/functions/fnc_deserializeState.sqf#L54

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- Title.
- Close #9636

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by johnb432** [comment]
```sqf
// Handle wound hashmaps deserialized as CBA_namespaces
if (!isNil "_value" && {_value isEqualType locationNull}) then {
    private _keys = allVariables _value;
    private _values = _keys apply {_value getVariable _x};
    _value = _keys createHashMapFromArray _values;
};

// Treat null as nil
if (isNil "_value" || {_value isEqualTo objNull}) then {
    _value = _default;
};
```

The suggested code above still throws an error in scheduled, so your fix is needed.

I added the `!isNil "_value"` and `isNil "_value"` checks, so that it's explicitly written. I also changed `typeName _value == "LOCATION"` to `_value isEqualType locationNull`.

**Comment by LinkIsGrim** [comment]
You're gonna have to ELI5 above, besides the `typeName` change

**Comment by johnb432** [comment]
> You're gonna have to ELI5 above, besides the `typeName` change

https://github.com/acemod/ACE3/blob/65be883eea0d22644f8cfbe12485e4295dbd31fa/addons/medical/functions/fnc_deserializeState.sqf#L59-L69

Replace these lines with the ones I put above.

Essentially, we now are doing `isNil` checks where there were none previously. `_value isEqualTo objNull`, with `_value` being `nil`, would not behave as expected.

**Comment by PabstMirror** [comment]
```
{
    private _value = _state getVariable _x;
    if (typeName _value == "LOCATION") then {
 
} forEach [
    [VAR_BLOOD_VOL, DEFAULT_BLOOD_VOLUME],
```
I'm pretty sure `_value` should always be defined
`_x` is an array not a string

**Review by PabstMirror — APPROVED** [review]
